src/components/Network.js

```diff
@@ -67,6 +67,28 @@ class Network extends React.Component {
     return null;
   }
 
+  getSyncV1Progress(sync_v1) {
+    const first_timestamp = this.state.dag.first_timestamp;
+    const latest_timestamp = this.state.dag.latest_timestamp;
+    const delta = latest_timestamp - first_timestamp;
+    const synced_percent = 100 * (sync_v1.synced_timestamp - first_timestamp) / delta;
+    let general_percent = (100 * (sync_v1.latest_timestamp - first_timestamp) / delta) - synced_percent;
+    if (general_percent > 100) {
+      general_percent = 100;
+    }
+    return { synced_percent, general_percent };
+  }
+
+  getSyncV2Progress(sync_v2) {
+    const synced_percent = Math.min(100, Math.abs(sync_v2.progress) * 100);
+    const general_percent = 100 - synced_percent;
+    return { synced_percent, general_percent };
+  }
+
+  getSyncProgress(sync_v1, sync_v2) {
```

Comment:
Maybe this method should receive `conn` directly and do the logic here (instead of in the method `renderConnected`). What do you think?

Reply:
Agreed. Done.